pkg/clients/clients_test.go

```diff
@@ -0,0 +1,72 @@
+package clients
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift-psap/special-resource-operator/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestPkgClients(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Clients Suite")
+}
+
+var _ = Describe("GetNodesByLabels", func() {
+	type testInput struct {
+		numNodes         int
+		labels           map[string]string
+		addTaint         bool
+		taintEffect      corev1.TaintEffect
+		expectedNumNodes int
+	}
+
+	DescribeTable(
+		"should return correct number of nodes",
+		func(test testInput) {
+			nodesList := utils.CreateNodesList(test.numNodes, test.labels)
+			if test.addTaint {
+				utils.AddTaint(&nodesList.Items[0], "taintKey", "taintValue", test.taintEffect)
+			}
+			objs := []runtime.Object{nodesList}
+			clientsStruct := k8sClients{runtimeClient: fake.NewFakeClient(objs...)}
```

Comment:
Why do you use the fake client here? I understood we wanted to mock all instead.

Reply:
we are testing the clients package, which directly uses runtime-client. I don't know any other way to mock a call to the runtime-client

Reply:
Oh indeed, sorry.